### ISSUE TITLE
Reproduce doesn't match result showed

### ISSUE THREAD

**Issue by zjw1990** [pytorch/fairseq#3001]
Hi,

I just reproduced id-en speech translation result and it shows the the result is:

Generate test_st_id_en with beam=5: BLEU = 10.77 61.7/19.6/6.2/2.5 (BP = 0.918 ratio = 0.921 hyp_len = 27497 ref_len = 29845)

without pre-train ASR.

But the result shows id-en BLEU should be 2.5. I didn't change and structures, so did I done sth wrong?

To reproduce:

python train.py ${COVO_ROOT} --train-subset train_st_id_en --valid-subset dev_st_id_en --save-dir ${ST_SAVE_DIR} \
    --num-workers 8 --max-tokens 8000 --task speech_to_text --criterion label_smoothed_cross_entropy \
    --report-accuracy --max-update 10000 --arch s2t_transformer_s  --optimizer adam --lr 2e-3 \
    --lr-scheduler inverse_sqrt --warmup-updates 10000 --clip-norm 10.0 --seed 1 --config-yaml ${COVO_ROOT}/config_st_ar_en.yaml

To evaluate:

python scripts/average_checkpoints.py --inputs ${ST_SAVE_DIR} --num-epoch-checkpoints 10 \
    --output "${ST_SAVE_DIR}/${CHECKPOINT_FILENAME}"

python generate.py ${COVO_ROOT} --gen-subset test_st_id_en --task task_speech_to_text_vae --batch-size 5 \
    --path ${ST_SAVE_DIR}/${CHECKPOINT_FILENAME} --beam 5 --scoring sacrebleu --config-yaml ${COVO_ROOT}/config_st_id_en.yaml
